lib/core/src/data/container.rs

```diff
@@ -0,0 +1,13 @@
+// ===========
+// === Add ===
+// ===========
+
+/// An abstraction for container which can be provided with new elements. The
+/// element type is polymorphic, allowing the container to reuse the function
+/// for different item types.
+pub trait Add<T> {
+    type Result = ();
+    fn add(&mut self, component:T) -> Self::Result;
```

Comment:
Ok, I don't like it anymore. So I'm pushing a new commit soon which implements `Extend` trait as well for our structures, but damn, the usage of `extend` is ugly. You want to write `world.add(workspace)` instead of something like `world.add(vec![workspace])` or any other artificial iterator here. Maybe we should rename it to `ExtendSingle`? On the other hand, writing `world.extend_single(workspace)` seems ugly as hell.

Reply:
The best is `std::iter::once` for this usage. You can just implement Extend for each implementor of Add